### PR TITLE
filter panel UI, add button to expand/collapse the search panel

### PR DESCRIPTION
**Pull request by Dav-C**
# Closes: #2185
# Add a filter button that collapses and expands the search panel. This button only appears if { filter_form } context has been provided.

# TODO

- [X] Explanation of Change(s)
- [X] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Desig
image1
image2

### CODE REVIEW

File: nautobot/core/templates/generic/object_list.html [hunk 1]
```diff
@@ -6,6 +6,9 @@
 {% block content %}
 <div class="pull-right noprint">
     {% block buttons %}{% endblock %}
+    {% if request.user.is_authenticated and filter_form %}
```
Comment (bryanculver) [inline nautobot/core/templates/generic/object_list.html:9]:
```suggestion
    {% if filter_form %}
```

**Review by bryanculver — CHANGES_REQUESTED** [review]
I'm curious how this plays into #1998

**Review by Dav-C — COMMENTED** [review]
I included request.user.is_authenticated to maintain parity with the config button, but I see its not really necessary.

**Comment by Dav-C** [comment]
Due to some upcoming UI changes, I'll close this PR for now and revisit after the next release.

**Pull request closed without merging**